bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BValueArray.java

```diff
@@ -0,0 +1,573 @@
+    }
+
+    public BValueArray(BType type) {
+        if (type.getTag() == TypeTags.INT_TAG) {
+            intValues = (long[]) newArrayInstance(Long.TYPE);
+            super.arrayType = new BArrayType(BTypes.typeInt);
+            this.elementType = BTypes.typeInt;
+        } else if (type.getTag() == TypeTags.BOOLEAN_TAG) {
+            booleanValues = (int[]) newArrayInstance(Integer.TYPE);
+            super.arrayType = new BArrayType(BTypes.typeBoolean);
+            this.elementType = BTypes.typeBoolean;
+        } else if (type.getTag() == TypeTags.BYTE_TAG) {
+            byteValues = (byte[]) newArrayInstance(Byte.TYPE);
+            super.arrayType = new BArrayType(BTypes.typeByte, size);
+            this.elementType = BTypes.typeByte;
+        } else if (type.getTag() == TypeTags.FLOAT_TAG) {
+            floatValues = (double[]) newArrayInstance(Double.TYPE);
+            super.arrayType = new BArrayType(BTypes.typeFloat);
+            this.elementType = BTypes.typeFloat;
+        } else if (type.getTag() == TypeTags.STRING_TAG) {
+            stringValues = (String[]) newArrayInstance(String.class);
+            Arrays.fill(stringValues, BLangConstants.STRING_EMPTY_VALUE);
+            super.arrayType = new BArrayType(BTypes.typeString);
+            this.elementType = BTypes.typeString;
+        } else {
+            super.arrayType = type;
+            if (type.getTag() == TypeTags.ARRAY_TAG) {
+                BArrayType arrayType = (BArrayType) type;
+                if (arrayType.getState() == BArrayState.CLOSED_SEALED) {
+                    this.size = maxArraySize = arrayType.getSize();
+                }
+                refValues = (BRefType[]) newArrayInstance(BRefType.class);
+                Arrays.fill(refValues, arrayType.getElementType().getZeroValue());
+            } else if (type.getTag() == TypeTags.TUPLE_TAG) {
+                BTupleType tupleType = (BTupleType) type;
+                this.size = maxArraySize = tupleType.getTupleTypes().size();
+                refValues = (BRefType[]) newArrayInstance(BRefType.class);
+                AtomicInteger counter = new AtomicInteger(0);
+                tupleType.getTupleTypes().forEach(memType ->
+                        refValues[counter.getAndIncrement()] = memType.getEmptyValue());
+            } else {
+                refValues = (BRefType[]) newArrayInstance(BRefType.class);
+                Arrays.fill(refValues, type.getEmptyValue());
+            }
+        }
+    }
+
+    public BValueArray() {
+        refValues = (BRefType[]) newArrayInstance(BRefType.class);
+    }
+
+    public BValueArray(BType type, int size) {
+        if (type.getTag() == TypeTags.INT_TAG) {
+            if (size != -1) {
+                this.size = maxArraySize = size;
+            }
+            intValues = (long[]) newArrayInstance(Long.TYPE);
+            super.arrayType = new BArrayType(BTypes.typeInt, size);
+            this.elementType = BTypes.typeInt;
```

Comment:
we can call the other constructor here by passing `(long[]) newArrayInstance(Long.TYPE);`. Applicable to other if blocks as well.